client/server/lib/memcached/index.ts

```diff
@@ -0,0 +1,47 @@
+import config from '@automattic/calypso-config';
+import { Client } from 'memjs';
+
+// Memcached should not be used in the client or environments where the server is
+// not configured.
+const MEMCACHED_DISABLE = typeof document !== 'undefined' || ! config( 'server_side_memcached' );
+let MEMCACHED_CLIENT: Client | undefined;
+
+const ENV = config( 'env_id' ) as string;
+
+export function createMemcachedClient() {
+	if ( MEMCACHED_DISABLE || MEMCACHED_CLIENT ) {
+		return;
+	}
+	// todo: do we need to handle disconnects/reconnects?
+	MEMCACHED_CLIENT = Client.create( 'foo:bar' ); // TODO: commit address or no?
+}
+
+/**
+ * Adds the environment to the memcached key removes whitespace.
+ *
+ * @param key The memcached key
+ * @returns The normalized memcached key
+ */
+function normalizeKey( key: string ): string {
+	return ( ENV + key ).replace( /\s/g, '_' );
```

Comment:
Should we address multiple consecutive spaces just in case?
```suggestion
	return ( ENV + key ).replace( /\s+/g, '_' );
```

Reply:
Sure! (This does appear to be handled already, but this approach is cleaner.) 

```js
console.log("hello  there".replace(/\s/g, '_'))
// hello__there

console.log("hello  there".replace(/\s+/g, '_'))
// hello_there
```